rmw_fastrtps_shared_cpp/src/rmw_publisher.cpp

```diff
@@ -50,41 +47,30 @@ __rmw_destroy_publisher(
     return RMW_RET_ERROR;
   }
 
-  if (!publisher) {
-    RMW_SET_ERROR_MSG("publisher handle is null");
-    return RMW_RET_ERROR;
-  }
-
-  if (publisher->implementation_identifier != identifier) {
-    RMW_SET_ERROR_MSG("publisher handle not from this implementation");
-    return RMW_RET_ERROR;
-  }
-
-  auto info = static_cast<CustomPublisherInfo *>(publisher->data);
-  if (info != nullptr) {
-    if (info->publisher_ != nullptr) {
-      Domain::removePublisher(info->publisher_);
-    }
-    if (info->listener_ != nullptr) {
-      delete info->listener_;
+  auto common_context = static_cast<rmw_dds_common::Context *>(node->context->impl->common);
+  auto info = static_cast<const CustomPublisherInfo *>(publisher->data);
+  {
+    // Update graph
+    std::lock_guard<std::mutex> guard(common_context->node_update_mutex);
+    rmw_dds_common::msg::ParticipantEntitiesInfo msg =
+      common_context->graph_cache.dissociate_writer(
+      info->publisher_gid, common_context->gid, node->name, node->namespace_);
```

Comment:
@ivanpauno indentation is atypical here.

Reply:
I agree it's atypical, but uncrustify complains if I add an extra tab in L57.
Do you have any suggestion to make identation less atypical while pleasing uncrustify?

Reply:
Can 55 and 56 not fit on the same line?

Reply:
I guess, not, you could do this:

```c++
    rmw_dds_common::msg::ParticipantEntitiesInfo msg;
    msg = common_context->graph_cache.dissociate_writer(
      ...
```

Reply:
Solved in https://github.com/ros2/rmw_fastrtps/pull/312/commits/ec4a3f3e5304881aa607dd1153d0815b7509b9a6.

Reply:
Won't this interfere with copy elision?

Reply:
> Won't this interfere with copy elision?

Probably yes, as copy-assignment can't be elided AFAIK.
I will revert the change and left the not nice identing.

Reply:
See https://github.com/ros2/rmw_fastrtps/pull/312/commits/e85759abf12d488f9a139ea3aab6efc66057e2b1.